### DISCUSSION TITLE
Submission and authentication

### DISCUSSION THREAD

**Discussion by cwittern** [Jinntec/Fore#122] [Q&A] [answered]
Digging deeper into Fore, I am now trying to submit a result back to my exist-db server. 

While I am logged in at the exist-db, calling a Fore html file does not include the login credentials. Posting with `<fx-submission/>` gives me a `401 Unauthorized` error, which is to be expected.

So my question is, how should I set things up so that the authentication from exist is carried over to Fore?
(In case this is important, I am developing by gradually modifying the `simple-tei-header` example).

Any help appreciated,  Chris

**Comment by JoernT** [comment]
@cwittern could you share your code please? or an example that reproduces the problem? It costs a lot of time to rebuild such a situation for me. 

You could either create a gist or attach files here.

We have setup an authorized eXist-db app but doing the login also ourselves which allows to keep the credentials and then use fx-header element to send authorization header. This works well but - as said - a concrete examples (also for others to learn from) would help to clarify that use case.

**Reply by cwittern** [reply]
Sure, here is the gist: https://gist.github.com/cwittern/034746923d7a33684f70a4b4c567b324

**Comment by JoernT** [comment, marked as answer]
@cwittern hard to say just from the html code. We already did several eXist-db apps with authentication and usually it just works. When you're authenticated your browser has the right cookie set. Any `fetch` operation Fore does will pick up that and send it along by default. So the reason might also lie in your controller.xql or the rights being set on target collections or request methods being accepted.... Would need the complete xar to debug.

One thing however - i've not used 'put' method so far. What do you get when using 'post'? 

One example of an eXist-db app with Fore and authentication is [Tuttle](https://github.com/eeditiones/tuttle) but this application also does the login part and using `fx-header` to set the authorization header. Another - non-public - app works the way you describe - without any further measures for login.

Good news - it will work - bad news: we have to debug it. Are you using OpenAPI (aka Roaster) for you app? This is a good way to debug as it allows you to check if your endpoint is fine ( via the api.html) and returns what's expected.

**Reply by cwittern** [reply]
Thank you, this information helps me understand what is going on.  And I indeed came across Roaster and will use it in my app.
